packages/pages/src/dev/server/ssr/getLocalData.ts

```diff
@@ -1,18 +1,40 @@
 import path from "path";
 import fs from "fs";
 import { readdir } from "fs/promises";
+import { findTemplateModuleInternal } from "./findTemplateModuleInternal.js";
+import { ViteDevServer } from "vite";
 
 const LOCAL_DATA_PATH = "localData";
 
+/**
+ * LocalDataManifest contains data parsed out from the filesystem that is then used
+ * to generate in the indexPage.
+ */
 export interface LocalDataManifest {
-  static: string[];
-  entity: Map<string, { entityId: string; slug: string }[]>;
+  static: {
+    // The featureName for a specific static template
+    featureName: string;
+    // The return value of the template's getPath()
+    staticURL: string;
+  }[];
+  entity: Map<
+    string,
+    {
+      // The entity's document.id
+      entityId: string;
+      // The entity's document.slug
+      slug: string | undefined;
```

Comment:
When is slug undefined?

Reply:
when you don't have a slug defined in your entity